src/mscorlib/src/System/Environment.cs

```diff
@@ -821,13 +821,6 @@ private static void SetEnvironmentVariableCore(string variable, string value)
             if (string.IsNullOrEmpty(value) || value[0] == '\0')
                 value = null;
 
-            if (AppDomain.IsAppXModel() && !AppDomain.IsAppXDesignMode())
-            {
-                // Environment variable accessors are not approved modern API.
-                // so we throw PlatformNotSupportedException.
```

Comment:
Seems reasonable to just remove all the special casing in anticipation of getting the env API approved, unless you object